adapters/Pangle/PangleAdapter/Bidding/GADPangleRTBInterstitialRenderer.m

```diff
@@ -0,0 +1,93 @@
+//
+//  GADPangleRTBInterstitialRenderer.m
+//  PangleAdapter
+//
+//  Created by bytedance on 2022/1/11.
+//
+
+#import "GADPangleRTBInterstitialRenderer.h"
+#import "GADMediationAdapterPangleConstants.h"
+#import <BUAdSDK/BUAdSDK.h>
+#import "GADMAdapterPangleUtils.h"
+
+@interface GADPangleRTBInterstitialRenderer()<BUFullscreenVideoAdDelegate>
+
+@end
+
+@implementation GADPangleRTBInterstitialRenderer {
+    /// The completion handler to call when the ad loading succeeds or fails.
+    GADMediationInterstitialLoadCompletionHandler _loadCompletionHandler;
+    /// The Pangle Interstitial ad.
+    BUFullscreenVideoAd *_fullScreenAdVideo;
+    /// An ad event delegate to invoke when ad rendering events occur.
+    id<GADMediationInterstitialAdEventDelegate> _delegate;
+}
+
+- (void)renderInterstitialForAdConfiguration:
+(nonnull GADMediationInterstitialAdConfiguration *)adConfiguration
+                           completionHandler:(nonnull GADMediationInterstitialLoadCompletionHandler)completionHandler {
+    _loadCompletionHandler = completionHandler;
+    NSString *slotId = adConfiguration.credentials.settings[GADMAdapterPanglePlacementID] ?: @"";\
+    if (PangleIsEmptyString(slotId)) {
+        NSError *error = GADMAdapterPangleErrorWithCodeAndDescription(GADPangleErrorSlotIdNil, @"placementid cannot be nil.");
+        _loadCompletionHandler(nil, error);
+        return;
+    }
+    _fullScreenAdVideo = [[BUFullscreenVideoAd alloc]initWithSlotID:slotId];
+    _fullScreenAdVideo.delegate = self;
+    if (![_fullScreenAdVideo respondsToSelector:@selector(setAdMarkup:)]) {
+        NSError *error = GADMAdapterPangleErrorWithCodeAndDescription(GADPangleErrorVersionLow, @"Pangle SDK version is too low");
+        _loadCompletionHandler(nil, error);
+        return;
+    }
+    [_fullScreenAdVideo setAdMarkup:adConfiguration.bidResponse];
+}
+
+#pragma mark--GADMediationInterstitialAd
```

Comment:
`#pragma mark - GADMediationInterstitialAd`

The same applies to below and other places.